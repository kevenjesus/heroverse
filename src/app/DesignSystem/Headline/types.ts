import { fontSizes } from "@/app/styles/guide"
import { HeadlineType, colorsType, fontSizeType, weightTye } from "@/app/styles/types"
import { ReactNode } from "react"

export const HeadlineTypes = {
    h1: fontSizes.heading1,
    h2: fontSizes.heading2,
    h3: fontSizes.heading3,
    h4: fontSizes.heading4
}

export interface HeadlineProps {
    children: ReactNode
    size?: fontSizeType
    weight?: weightTye
    color?: colorsType
    as: HeadlineType
}