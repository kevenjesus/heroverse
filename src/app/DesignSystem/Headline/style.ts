import { colors, fontSizes, weightValues } from '@/app/styles/guide'
import { FontStyleHeadlineTYpe } from '@/app/styles/types'
import styled from 'styled-components'
import { HeadlineTypes } from './types'

export const HeadlineItem = styled.h1<FontStyleHeadlineTYpe>`
    line-height: 175%;
    color: ${({color}) => color ? color : colors.black};
    font-weight: ${({weight}) => weight ? weight : weightValues.bold};
    font-size: ${({type}) => type ? HeadlineTypes[type] : fontSizes.heading1};
`