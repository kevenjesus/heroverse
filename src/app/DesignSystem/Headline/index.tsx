import * as S from './style'
import { HeadlineProps } from './types'

export default function Headline({children, as, ...rest}: HeadlineProps) {
    return <S.HeadlineItem type={as} as={as} {...rest}>{children}</S.HeadlineItem>
}