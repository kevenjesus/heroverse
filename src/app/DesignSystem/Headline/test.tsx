import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import Headline from '.';

describe('Headline', () => {
  it('should render', () => {
    const { container } = render(<Headline as="h1">titulo</Headline>);
    expect(container).toMatchSnapshot()
  });

  it('should render headline specific tag', () => {
    const { container, getByText } = render(<Headline as="h2">titulo</Headline>);
    expect(container.getElementsByTagName('h2').length).toBe(1)
    expect(getByText('titulo')).toHaveStyle({'font-size': '25px'})
  });

  it('should render headline props', () => {
    const { getByText } = render(<Headline as="h1" color='darkBlue' weight='700'>titulo</Headline>);
    expect(getByText('titulo')).toHaveStyle({'font-weight': '700'})
  });
});
