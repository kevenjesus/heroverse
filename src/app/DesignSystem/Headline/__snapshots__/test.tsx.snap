// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline should render 1`] = `
<div>
  <h1
    class="sc-dmqHEX mdbIu"
    type="h1"
  >
    titulo
  </h1>
</div>
`;
